src/components/navigation/BucketButton.js

```diff
@@ -0,0 +1,124 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../../util/typing';
+
+import ButtonBase from '../input/ButtonBase';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {import('../input/ButtonBase').HTMLButtonAttributes} HTMLButtonAttributes
+ * @typedef {import('../input/ButtonBase').ButtonCommonProps} ButtonCommonProps
+ *
+ * @typedef BucketButtonProps
+ * @prop {'left'|'up'|'down'|'lozenge'} [variant='lozenge']
```

Comment:
I agonized on this a bit. `variant` is a prop name consistent with other buttons and presentational components, but `direction` sure feels more natural...